src/ko/wolfdotcom/src/eu/kanade/tachiyomi/extension/ko/wolfdotcom/Wolf.kt

```diff
@@ -0,0 +1,398 @@
+            filters.filterIsInstance<UrlPartFilter>().forEach { filter ->
+                filter.addToUrl(this)
+            }
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    private fun parseBrowsePage(response: Response) {
+        val document = response.asJsoup()
+
+        parseSearchFilters(document)
+
+        browseCache = document.select(".webtoon-list > ul > li > a").map {
+            val id = it.absUrl("href").toHttpUrl()
+                .queryParameter("toon")!!.toInt()
+
+            BrowseItem(
+                id = id,
+                title = it.selectFirst(".txt > .subject")!!.ownText(),
+                cover = it.selectFirst(".img > img")?.attr("data-original"),
+            )
+        }.chunked(20)
+    }
+
+    private fun paginatedBrowsePage(index: Int): MangasPage {
+        return MangasPage(
+            browseCache[index].map {
+                SManga.create().apply {
+                    url = it.id.toString()
+                    title = it.title
+                    thumbnail_url = it.cover
+                }
+            },
+            browseCache.lastIndex > index,
+        )
+    }
+
+    private val specialChars = Regex("""[^가-힣0-9a-zA-Z ]""", RegexOption.IGNORE_CASE)
+    private val styleImage = Regex("""background-image:url\(([^)]+)\)""")
+
+    private fun querySearch(query: String): Observable<MangasPage> {
+        if (query.length < 2) {
+            throw Exception("두 글자 이상 입력 해주세요.")
+        }
+        val stdQuery = query.replace(specialChars, "")
+        val searchUrl = "$baseUrl/search.html?q=${URLEncoder.encode(stdQuery, "EUC-KR")}"
+
+        return client.newCall(GET(searchUrl, headers))
+            .asObservableSuccess()
+            .map { response ->
+                val document = Jsoup.parseBodyFragment(response.body.string(), searchUrl)
+                val entries = document.select("article.searchItem")
+                    .filter { el ->
+                        el.selectFirst("a.searchLink")!!.attr("href").contains(entryPath)
+                    }
+                    .map { el ->
+                        val mangaUrl = el.selectFirst("a.searchLink")!!.absUrl("href")
+                            .toHttpUrl()
+                        SManga.create().apply {
+                            url = mangaUrl.queryParameter("toon")!!
+                            title = el.selectFirst(".searchDetailTitle")!!.text()
+                            thumbnail_url = el.selectFirst(".searchPng")
+                                ?.attr("style")
+                                ?.let {
+                                    styleImage.find(it)?.groupValues?.get(1)
+                                }
+                        }
+                    }
+
+                MangasPage(entries, false)
+            }
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/$entryPath?toon=${manga.url}"
```

Comment:
It might make sense to use a `HttpUrlBuilder`, for escaping.

Reply:
`manga.url` have an integer id stored so should be fine